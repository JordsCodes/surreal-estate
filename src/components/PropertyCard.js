import React from "react";
import "../styles/property-card.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBed,
  faCity,
  faEnvelope,
  faHouse,
  faPoundSign,
  faToilet,
} from "@fortawesome/free-solid-svg-icons";

const PropertyCard = ({
  title,
  city,
  type,
  bedrooms,
  bathrooms,
  price,
  email,
}) => {
  return (
    <div className="property-card">
      <p className="property-card_title">{title}</p>
      <p>
        <FontAwesomeIcon icon={faCity} /> &nbsp;
        {city}
      </p>
      <p>
        <FontAwesomeIcon icon={faPoundSign} /> &nbsp;{price}
      </p>
      <p>
        <FontAwesomeIcon icon={faHouse} />
        &nbsp;{type}
      </p>
      <p>
        {bedrooms}&nbsp; <FontAwesomeIcon icon={faBed} /> &nbsp;&nbsp;
        {bathrooms}&nbsp; <FontAwesomeIcon icon={faToilet} />
      </p>
      <p>
        <FontAwesomeIcon icon={faEnvelope} /> &nbsp;
        {email}
      </p>
    </div>
  );
};

export default PropertyCard;
