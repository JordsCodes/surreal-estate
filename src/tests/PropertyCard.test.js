import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("Property Card", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<PropertyCard />);
    const propertyCard = asFragment();

    expect(propertyCard).toMatchSnapshot();
  });

  it("renders correct value for title", () => {
    const component = render(<PropertyCard title="4 Bedroom Townhouse" />);
    const title = component.getByText("4 Bedroom Townhouse");

    expect(title.textContent).toBe("4 Bedroom Townhouse");
  });

  it("renders correct value for city", () => {
    const component = render(<PropertyCard city="Newcastle upon Tyne" />);
    const city = component.getByText("Newcastle upon Tyne");

    expect(city.textContent).toContain("Newcastle upon Tyne");
  });

  it("renders correct value for price", () => {
    const component = render(<PropertyCard price="300000" />);
    const price = component.getByText("300000");

    expect(price.textContent).toContain("300000");
  });

  it("renders correct value for type", () => {
    const component = render(<PropertyCard type="terrace" />);
    const type = component.getByText("terrace");

    expect(type.textContent).toContain("terrace");
  });

  it("renders correct value for bedrooms", () => {
    const component = render(<PropertyCard bedrooms="4" />);
    const bedrooms = component.getByText("4");

    expect(bedrooms.textContent).toContain("4");
  });

  it("renders correct value for bathrooms", () => {
    const component = render(<PropertyCard bathrooms="2" />);
    const bathrooms = component.getByText("2");

    expect(bathrooms.textContent).toContain("2");
  });

  it("renders correct value for email", () => {
    const component = render(<PropertyCard email="jordan@toon.co.uk" />);
    const email = component.getByText("jordan@toon.co.uk");

    expect(email.textContent).toContain("jordan@toon.co.uk");
  });
});
