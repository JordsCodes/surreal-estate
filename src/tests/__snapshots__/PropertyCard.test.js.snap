// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <p
      class="property-card_title"
    />
    <p>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-city "
        data-icon="city"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M480 48c0-26.5-21.5-48-48-48H336c-26.5 0-48 21.5-48 48V96H224V24c0-13.3-10.7-24-24-24s-24 10.7-24 24V96H112V24c0-13.3-10.7-24-24-24S64 10.7 64 24V96H48C21.5 96 0 117.5 0 144v96V464c0 26.5 21.5 48 48 48H304h32 96H592c26.5 0 48-21.5 48-48V240c0-26.5-21.5-48-48-48H480V48zm96 320v32c0 8.8-7.2 16-16 16H528c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16zM240 416H208c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16zM128 400c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V368c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32zM560 256c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H528c-8.8 0-16-7.2-16-16V272c0-8.8 7.2-16 16-16h32zM256 176v32c0 8.8-7.2 16-16 16H208c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16zM112 160c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h32zM256 304c0 8.8-7.2 16-16 16H208c-8.8 0-16-7.2-16-16V272c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32zM112 320H80c-8.8 0-16-7.2-16-16V272c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16zm304-48v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V272c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16zM400 64c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V80c0-8.8 7.2-16 16-16h32zm16 112v32c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16V176c0-8.8 7.2-16 16-16h32c8.8 0 16 7.2 16 16z"
          fill="currentColor"
        />
      </svg>
        
    </p>
    <p>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sterling-sign "
        data-icon="sterling-sign"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M112 160.4c0-35.5 28.8-64.4 64.4-64.4c6.9 0 13.8 1.1 20.4 3.3l81.2 27.1c16.8 5.6 34.9-3.5 40.5-20.2s-3.5-34.9-20.2-40.5L217 38.6c-13.1-4.4-26.8-6.6-40.6-6.6C105.5 32 48 89.5 48 160.4V224H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H48v44.5c0 17.4-4.7 34.5-13.7 49.4L4.6 431.5c-5.9 9.9-6.1 22.2-.4 32.2S20.5 480 32 480H288c17.7 0 32-14.3 32-32s-14.3-32-32-32H88.5l.7-1.1C104.1 390 112 361.5 112 332.5V288H224c17.7 0 32-14.3 32-32s-14.3-32-32-32H112V160.4z"
          fill="currentColor"
        />
      </svg>
        
    </p>
    <p>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-house "
        data-icon="house"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1V472c0 22.1-17.9 40-40 40H456c-1.1 0-2.2 0-3.3-.1c-1.4 .1-2.8 .1-4.2 .1H416 392c-22.1 0-40-17.9-40-40V448 384c0-17.7-14.3-32-32-32H256c-17.7 0-32 14.3-32 32v64 24c0 22.1-17.9 40-40 40H160 128.1c-1.5 0-3-.1-4.5-.2c-1.2 .1-2.4 .2-3.6 .2H104c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9 .1-2.8V287.6H32c-18 0-32-14-32-32.1c0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7L564.8 231.5c8 7 12 15 11 24z"
          fill="currentColor"
        />
      </svg>
       
    </p>
    <p>
        
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed "
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 32c17.7 0 32 14.3 32 32V320H288V160c0-17.7 14.3-32 32-32H544c53 0 96 43 96 96V448c0 17.7-14.3 32-32 32s-32-14.3-32-32V416H352 320 64v32c0 17.7-14.3 32-32 32s-32-14.3-32-32V64C0 46.3 14.3 32 32 32zm144 96a80 80 0 1 1 0 160 80 80 0 1 1 0-160z"
          fill="currentColor"
        />
      </svg>
           
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-toilet "
        data-icon="toilet"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 0C10.7 0 0 10.7 0 24S10.7 48 24 48h8V196.9c-1.9 1.4-3.8 2.9-5.6 4.4C10.9 214.5 0 232.9 0 256c0 46.9 14.3 84.1 37 112.5c14.2 17.7 31.1 31.3 48.5 41.8L65.6 469.9c-3.3 9.8-1.6 20.5 4.4 28.8s15.7 13.3 26 13.3H352c10.3 0 19.9-4.9 26-13.3s7.7-19.1 4.4-28.8l-19.8-59.5c17.4-10.5 34.3-24.1 48.5-41.8c22.7-28.4 37-65.5 37-112.5c0-23.1-10.9-41.5-26.4-54.6c-1.8-1.5-3.7-3-5.6-4.4V48h8c13.3 0 24-10.7 24-24s-10.7-24-24-24H24zM384 256.3c0 1-.3 2.6-3.8 5.6c-4.8 4.1-14 9-29.3 13.4C320.5 284 276.1 288 224 288s-96.5-4-126.9-12.8c-15.3-4.4-24.5-9.3-29.3-13.4c-3.5-3-3.8-4.6-3.8-5.6l0-.3 0-.1c0-1 0-2.5 3.8-5.8c4.8-4.1 14-9 29.3-13.4C127.5 228 171.9 224 224 224s96.5 4 126.9 12.8c15.3 4.4 24.5 9.3 29.3 13.4c3.8 3.2 3.8 4.8 3.8 5.8l0 .1 0 .3zM328.2 384l-.2 .5 0-.5h.2zM112 64h32c8.8 0 16 7.2 16 16s-7.2 16-16 16H112c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
        />
      </svg>
    </p>
    <p>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-envelope "
        data-icon="envelope"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
          fill="currentColor"
        />
      </svg>
        
    </p>
  </div>
</DocumentFragment>
`;
